lib/rules/value-no-vendor-prefix/index.mjs

```diff
@@ -52,8 +53,23 @@ const rule = (primary, secondaryOptions, context) => {
 				return;
 			}
 
-			if (optionsMatches(secondaryOptions, 'ignoreValues', vendor.unprefixed(value))) {
-				return;
+			if (secondaryOptions?.ignoreValues) {
+				const isIgnored = secondaryOptions.ignoreValues.some(
+					(/** @type { string | RegExp } */ element) => {
+						const isStringifiedRegex =
+							isString(element) && element[0] === '/' && element.at(-1) === '/';
+						const useRawValue = element instanceof RegExp || isStringifiedRegex;
+						/**
+						 * @todo consolidate in the next major
+						 * @see stylelint/stylelin#7542
+						 */
+						const input = useRawValue ? value : vendor.unprefixed(value);
+
+						return Boolean(testAgainstStringOrRegExp(input, element));
```

Comment:
[suggest] I find a way to avoid exporting `testAgainstStringOrRegExp`. This uses `optionsMatches()` as before. How about this?

```js
		if (!validOptions) {
			return;
		}

		/** @type {{ raw: Array<string | RegExp>, unprefixed: Array<string | RegExp> }} */
		const ignoreValues = { raw: [], unprefixed: [] };

		secondaryOptions?.ignoreValues?.forEach((element) => {
			const useRawValue =
				isRegExp(element) ||
				(element.startsWith('/') && (element.endsWith('/') || element.endsWith('/i')));

			if (useRawValue) {
				ignoreValues.raw.push(element);
			} else {
				ignoreValues.unprefixed.push(element);
			}
		});

		root.walkDecls((decl) => {
			// ...

			if (optionsMatches(ignoreValues, 'raw', value)) {
				return;
			}

			if (optionsMatches(ignoreValues, 'unprefixed', vendor.unprefixed(value))) {
				return;
			}

			// ...
		});
```